enhancements/ingress/logging-api.md

```diff
@@ -0,0 +1,370 @@
+
+The [cluster logging log forwarding
+API](../cluster-logging/cluster-logging-log-forwarding.md) may eventually
+supersede elements of the Ingress logging API, so it is important to design the
+Ingress logging API with the log forwarding API in mind in order to avoid
+redundancy or other issues.
+
+## Design Details
+
+### Test Plan
+
+The controller that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  The operator has end-to-end
+tests; for this enhancement, a test is added that (1) configures a pod with
+rsyslog, (2) creates an IngressController configured to send access logs to this
+pod, and (3) verifies that the rsyslog pod receives access logs.
+
+### Graduation Criteria
+
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, access logging is disabled by default, which is consistent with the
+feature's absence in older versions.  On downgrade, the IngressController
+Deployment is updated, and any sidecar and `EmptyDir` volume are automatically
+removed from the Deployment.  A harmless ConfigMap with rsyslog configuration
+may remain (with an owner reference on the Deployment).
+
+### Version Skew Strategy
+
+N/A.
+
+## Implementation History
+
+In OpenShift 1.3/3.3, the `ROUTER_SYSLOG_ADDRESS` environment variable [was
+added](https://github.com/openshift/origin/pull/8332/commits/311abc322d69d63c5dd03ec3dfbef7f88d039546),
+and in OpenShift 1.5/3.5, the `ROUTER_LOG_FACILITY` environment variable [was
+added](https://github.com/openshift/origin/pull/12795/commits/c01fab1b76388e213f595ccc99dcbfdc5d1f1401);
+these environment variables are used to implement the logging API.
+
+In OpenShift 3.11, the `oc adm router --extended-logging` flag [was
+added](https://github.com/openshift/origin/pull/20260/commits/9d6a4c2cfc6a03137dd07d645a6c7db73b0e15aa),
+and in OpenShift 4.2, the `ingress.operator.openshift.io/unsupported-logging`
+annotation for IngressController and Ingress config [was
+added](https://github.com/openshift/cluster-ingress-operator/pull/293/commits/5a82b658c4952e8af5d4658d13e8802b93ded870);
+some of this logic from `oc adm router` is re-used in the ingress operator, and
+the logging API in part formalizes the unsupported annotation.
+
+Finally,
+[openshift/cluster-ingress-operator#374](https://github.com/openshift/cluster-ingress-operator/pull/374) implements this enhancement.
+
+## Alternatives
+
+### Cluster Logging Log Forwarding API
+
+Instead of expanding the IngressController API, the [the cluster logging log
+forwarding API](../cluster-logging/cluster-logging-log-forwarding.md) could be
+expanded to encompass the necessary configuration.
+
+#### Drawbacks
+
+The current cluster logging log forwarding API does not support writing straight
+to a syslog endpoint.
+
+Configuring logging for individual components is outside the scope of the
+cluster logging log forwarding API.
```

Comment:
@Miciah IMO configuring logging in one place for all platform components is preferred. From a quick review of the cluster logging [EP](https://github.com/openshift/enhancements/blob/master/enhancements/cluster-logging/cluster-logging-log-forwarding.md), it attempts to provide a cluster-wide log fwd'ing solution for apps, infra and audit:

> Selectively forward Application, Infrastructure and Audit inputs.

If this is the case, then why not use the cluster logging API? If the EP does not meet our requirements, then why not work with that team to fix their API? For example, both EP's provide an interface to configure standard Syslog parameters.